junit-jupiter-params/src/main/java/org/junit/jupiter/params/ResolutionCache.java

```diff
@@ -36,7 +36,7 @@ class Concurrent implements ResolutionCache {
 		private final Map<ParameterDeclaration, Object> cache = new ConcurrentHashMap<>();
 
 		@Override
-		public Object resolve(ParameterDeclaration declaration, Supplier<Object> resolver) {
+		public @Nullable Object resolve(ParameterDeclaration declaration, Supplier<@Nullable Object> resolver) {
```

Comment:
Should it be `Supplier<? extends @Nullable Object>`?

I am afraid the current type signature does not accept `Supplier<@NonNull Object>` suppliers.

Reply:
Assertions have a lot of `Supplier<@Nullable String>` code which would be incompatible with `Supplier<String>` values :-/